java/ql/lib/semmle/code/java/security/SensitiveKeyboardCacheQuery.qll

```diff
@@ -48,5 +131,13 @@ private predicate inputTypeCached(string ty) {
 /** Gets an input field whose contents may be sensitive and may be stored in the keyboard cache. */
 AndroidEditableXmlElement getASensitiveCachedInput() {
   result.getId().regexpMatch(getInputSensitiveInfoRegex()) and
-  inputTypeCached(result.getInputType())
+  (
+    inputTypeCached(result.getInputType())
+    or
+    not exists(result.getInputType()) and
+    not exists(GoodInputTypeConf conf, DataFlow::Node src, DataFlow::Node sink |
+      conf.hasFlow(src, sink) and
+      sink.asExpr() = setInputTypeForId(result.getId())
```

Comment:
Does setting the input type in source code only work if it hasn't been set in XML? (i.e. setting it in XML has preference over source code)